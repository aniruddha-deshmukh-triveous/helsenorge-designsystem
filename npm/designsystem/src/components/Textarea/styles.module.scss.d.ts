export type Styles = {
  'content-wrapper': string;
  'content-wrapper__input': string;
  'content-wrapper__input--disabled': string;
  'content-wrapper--bigform': string;
  'content-wrapper--disabled': string;
  'content-wrapper--invalid': string;
  'content-wrapper--on-blueberry': string;
  'content-wrapper--on-dark': string;
  'content-wrapper--transparent': string;
  textarea: string;
  'textarea__after-label-children': string;
  'textarea__counter-wrapper': string;
  'textarea__counter-wrapper--invalid': string;
  'textarea__counter-wrapper--on-dark': string;
  'textarea__error-text': string;
  'textarea__label-wrapper': string;
  'textarea__label-wrapper--on-dark': string;
  'textarea--gutterBottom': string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
