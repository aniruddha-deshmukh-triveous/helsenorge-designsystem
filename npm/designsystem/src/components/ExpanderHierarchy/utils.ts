export type HeadingTags = 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

export const getHeadingTag = (htmlMarkup?: HeadingTags): HeadingTags => {
  switch (htmlMarkup) {
    case 'h2':
      return 'h3';
    case 'h3':
      return 'h4';
    case 'h4':
      return 'h5';
    case 'h5':
      return 'h6';
    case 'h6':
      return 'h6';
    default:
      return 'h2';
  }
};
