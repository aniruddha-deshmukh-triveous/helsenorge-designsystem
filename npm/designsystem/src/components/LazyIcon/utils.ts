export const isServerSide = (): boolean => typeof document === 'undefined';
