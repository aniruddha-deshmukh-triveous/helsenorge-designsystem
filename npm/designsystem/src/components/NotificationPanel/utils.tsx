import React, { AriaAttributes } from 'react';
import { palette } from '../../theme/palette';
import Icon from '../Icons';

import InfoSignStroke from '../Icons/InfoSignStroke';
import AlertSignStroke from '../Icons/AlertSignStroke';
import AlertSignFill from '../Icons/AlertSignFill';

export const variantToIconMap = {
  info: <Icon svgIcon={InfoSignStroke} color={palette.kiwi900} hoverColor={palette.kiwi900} />,
  warn: <Icon svgIcon={AlertSignStroke} color="black" hoverColor="black" />,
  alert: <Icon svgIcon={AlertSignFill} color={palette.cherry500} hoverColor={palette.cherry500} />,
  alertLabel: <Icon svgIcon={AlertSignStroke} color={palette.cherry500} hoverColor={palette.cherry500} />,
  crisis: <Icon svgIcon={AlertSignFill} color={palette.banana200} hoverColor={palette.banana200} />,
};

interface AriaLabelAttributesConfig {
  label?: string;
  id?: string;
}

type AriaLabelAttributes = Pick<AriaAttributes, 'aria-label' | 'aria-labelledby'>;

export const getAriaLabelAttributes = (config: AriaLabelAttributesConfig): AriaLabelAttributes | undefined => {
  const { label, id } = config;

  if (id) {
    return {
      'aria-labelledby': id,
    };
  }
  if (label) {
    return {
      'aria-label': label,
    };
  }
};
