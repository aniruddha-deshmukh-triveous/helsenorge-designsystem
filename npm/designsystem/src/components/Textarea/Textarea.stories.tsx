import React from 'react';
import { withKnobs, text, select, number, boolean } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import Textarea from './Textarea';
import { withA11y } from '@storybook/addon-a11y';
import { mapToBackgoundColor } from '../../../.storybook/StoryBackground';

import { FormMode } from '../../constants';
import Hospital from '../Icons/Hospital';
import Icon, { IconSize } from '../Icons';

const stories = storiesOf('Textarea', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withA11y);

stories.add('Skeleton', () => (
  <div style={{ width: '20rem' }}>
    <Textarea
      placeholder="Dette er en placeholder"
      maxCharacters={150}
      maxText={'chars'}
      defaultValue={`It is not the fart that kills you, it's the smell \n \n
  - Petter Solberg`}
      testId="123-test"
      transparent={boolean('transparent', false)}
      disabled={boolean('disabled', false)}
      label="Skriv inn din tekst"
      minRows={number('minRows', 3)}
      maxRows={number('maxRows', 15)}
      grow={boolean('grow', true)}
      errorText={text('errorText', '')}
    />
  </div>
));

stories.add('All props', () => {
  const modes = select('background', FormMode, FormMode.onwhite);

  return (
    <div style={{ width: '80vw', background: mapToBackgoundColor(modes), padding: 20 }}>
      <Textarea
        maxCharacters={150}
        maxText={'chars'}
        defaultValue={`It is not the fart that kills you, it's the smell \n \n
  - Petter Solberg`}
        testId="123-test"
        transparent={boolean('transparent', false)}
        mode={modes}
        disabled={boolean('disabled', false)}
        label="Skriv inn din tekst"
        minRows={number('minRows', 3)}
        maxRows={number('maxRows', 15)}
        grow={boolean('grow', true)}
        errorText={text('errorText', '')}
      />
    </div>
  );
});

stories.add('Max characters', () => (
  <div style={{ paddingTop: 50 }}>
    <div style={{ display: 'flex', width: '50rem' }}>
      <Textarea maxCharacters={10} marginBottom />
    </div>

    <div style={{ display: 'flex', width: '40rem' }}>
      <Textarea maxCharacters={100} />
    </div>

    <div style={{ width: '20rem' }}>
      <Textarea maxCharacters={100} defaultValue="test" />
    </div>
  </div>
));

stories.add('Disabled', () => (
  <>
    <div style={{ width: '20rem' }}>
      <Textarea disabled placeholder={'This is a placeholder'} marginBottom />
    </div>
    <div style={{ width: '20rem' }}>
      <Textarea disabled defaultValue={`This is a default value`}></Textarea>
    </div>
  </>
));

stories.add('Transparent', () => (
  <div style={{ width: '25rem', background: '#e4f7f9' }}>
    <Textarea
      maxCharacters={150}
      transparent
      defaultValue={`It is not the fart that kills you, it's the smell
- Petter Solberg`}
    />
  </div>
));

stories.add('Children after label', () => (
  <div style={{ width: '25rem', background: '#e4f7f9' }}>
    <Textarea
      label={'Here is a child next to me'}
      maxCharacters={150}
      defaultValue={`It is not the fart that kills you, it's the smell
- Petter Solberg`}
      afterLabelChildren={<Icon size={IconSize.XSmall} svgIcon={Hospital}></Icon>}
    />
  </div>
));
