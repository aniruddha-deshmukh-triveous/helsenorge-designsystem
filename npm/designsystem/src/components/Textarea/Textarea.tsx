import React, { useState } from 'react';
import cn from 'classnames';

import styles from './styles.module.scss';
import { HTMLTextareaProps, ModeVariant } from '../../constants';
import uuid from '../../utils/uuid';
import { useEffect } from 'react';
import { useRef } from 'react';

interface TextareaProps extends HTMLTextareaProps {
  /** initial value for textarea */
  defaultValue?: string;
  /** max character limit in textarea  */
  max?: number;
  /** The text is displayed in the end of the text-counter */
  maxText?: string;
  /** Sets the data-testid attribute. */
  testId?: string;
  /** If true, the component will have a bottom margin. */
  gutterBottom?: boolean;
  /** If true, the component will be transparent. */
  transparent?: boolean;
  /** Changes the visuals of the textarea */
  mode?: keyof typeof ModeVariant;
  /** Label of the input */
  label?: string;
  /** max rows */
  maxRows?: number;
  /** min rows */
  minRows?: number;
  /** auto-grows until maxRows */
  grow?: boolean;
  /** Error text to show above the component */
  errorText?: string;
}

const Textarea = React.forwardRef((props: TextareaProps, ref: React.Ref<HTMLTextAreaElement>) => {
  const {
    max,
    maxText,
    testId,
    defaultValue,
    gutterBottom,
    transparent,
    mode,
    label,
    minRows = 3,
    maxRows = 10,
    grow,
    errorText,
    ...restProps
  } = props;

  const [rows, setRows] = useState(minRows);
  const [textareaInput, setTextareaInput] = useState(defaultValue || '');
  const referanse = useRef<HTMLDivElement>(null);

  const resizeHeight = (target: HTMLTextAreaElement): void => {
    const textareaLineHeight = 28;

    const previousRows = target.rows;
    target.rows = minRows; // reset number of rows in textarea

    const currentRows = Math.floor((target.scrollHeight - 16) / textareaLineHeight); // scrollHeight - 16px of padding to calculate the rows

    if (currentRows === previousRows) {
      target.rows = currentRows;
    }

    if (currentRows >= maxRows) {
      target.rows = maxRows;
      target.scrollTop = target.scrollHeight;
    }

    if (currentRows < maxRows) {
      setRows(currentRows);
    } else {
      setRows(maxRows);
    }
  };

  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>): void => {
    if (grow) {
      resizeHeight(event.target);
    }
    setTextareaInput(event.target.value);
  };

  const onDark = mode === ModeVariant.onDark;
  const onBlueberry = mode === ModeVariant.onBlueberry;
  const textHasError = max && textareaInput.length > max;
  const onError = mode === ModeVariant.onError || errorText || textHasError;

  const textareaWrapperClass = cn(styles.textarea, {
    [styles['textarea--gutterBottom']]: gutterBottom,
    [styles[`textarea--${ModeVariant.onError}`]]: errorText,
  });

  const textareaClass = cn(styles.textarea__input, {
    [styles['textarea__input--transparent']]: transparent,
    [styles[`textarea__input--${ModeVariant.onBlueberry}`]]: onBlueberry,
    [styles[`textarea__input--${ModeVariant.onDark}`]]: onDark,
    [styles[`textarea__input--${ModeVariant.onError}`]]: onError,
  });

  const counterTextClass = cn(styles['textarea__counter-wrapper'], {
    [styles[`textarea__counter-wrapper--${ModeVariant.onDark}`]]: onDark,
    [styles[`textarea__counter-wrapper--${ModeVariant.onError}`]]: onError,
  });

  const labelClass = cn(styles.textarea__label, {
    [styles[`textarea__label--${ModeVariant.onDark}`]]: onDark,
  });

  const uniqueId = label ? uuid() : undefined;

  useEffect(() => {
    if (grow && referanse.current?.children && referanse.current?.children[0]) {
      const textarea = referanse.current?.children[0] as HTMLTextAreaElement;
      resizeHeight(textarea);
    }
  }, []);

  return (
      <div data-testid={testId} className={textareaWrapperClass}>
        {label && (
          <div className={labelClass}>
            <label htmlFor={uniqueId}>{label}</label>
          </div>
        )}
        <div ref={referanse}>
          <textarea
            rows={rows}
            defaultValue={defaultValue}
            id={uniqueId}
            className={textareaClass}
            ref={ref}
            onChange={handleChange}
            {...restProps}
          />
          {max && (
            <div className={counterTextClass}>
              <p>{`${textareaInput.length}/${max} ${maxText ? maxText : 'tegn'}`}</p>
            </div>
          )}
        </div>
      </div>
  );
});

export default Textarea;
