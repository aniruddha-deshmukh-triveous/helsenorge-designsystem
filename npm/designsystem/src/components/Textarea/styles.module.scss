@import '../../scss/_spacers.scss';
@import '../../scss/_breakpoints.scss';
@import '../../scss/_palette.scss';
@import '../../scss/_font-settings.scss';
@import '../../scss/_input.scss';

.textarea {
  @include input-wrapper;

  &__error-text {
    font-size: $font-size-sm;
    line-height: $font-size-sm;
    font-weight: 600;
    color: $cherry500;
    margin-top: 0;
  }

  &--gutterBottom {
    margin-bottom: getSpacer(s);
  }

  &__counter-wrapper {
    display: flex;
    justify-content: end;
    padding-top: getSpacer(xs);
    padding-right: 2px;
    font-size: $font-size-sm;
    line-height: 28px;

    @media (min-width: map-get($grid-breakpoints, md)) {
      line-height: 30px;
      font-size: $font-size-md;
    }

    p {
      margin: 0;
    }

    &--on-dark {
      color: $white;
    }
    &--invalid {
      color: $cherry500;
      font-weight: 600;
    }
  }
}

.content-wrapper {
  @include content-wrapper;
  &__input {
    line-height: 28px;
    height: auto;
    padding: 16px;
    resize: none;
  }
}
