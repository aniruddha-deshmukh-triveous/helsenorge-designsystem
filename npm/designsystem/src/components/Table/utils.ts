import { Breakpoint } from '../../hooks/useBreakpoint';
import { isTouchDevice } from '../../utils/device';
import { BreakpointConfig, SmallViewportVariant } from './Table';

import styles from './styles.module.scss';

/**
 * Sjekk om det skal brukes CSS for å bestemme responsivt utseende på dette breakpoint
 * @param config Konfigurasjon for responsiv oppførsel
 * @returns true om breakpoint bruker CSS
 */
const configUsesCss = (config: BreakpointConfig): boolean =>
  config.variant === SmallViewportVariant.centeredoverflow || config.variant === SmallViewportVariant.block;

/**
 * Lag klassenavn for CSS-config
 * @param config Konfigurasjon for responsiv oppførsel
 * @returns CSS-klassenavn
 */
const mapConfigToClass = (config: BreakpointConfig): string => styles[`table--${config.variant}-${config.breakpoint}`];

/**
 * Sorter konfigurasjon etter breakpoints, fra største til minste
 * @param a Konfigurasjon for responsiv oppførsel
 * @param b Konfigurasjon for responsiv oppførsel
 * @returns Sortert liste
 */
const sortByBreakpointsDescending = (a: BreakpointConfig, b: BreakpointConfig): number =>
  Breakpoint[a.breakpoint] - Breakpoint[b.breakpoint];

/**
 * Sjekk om en konfigurasjon skal brukes for nåværende breakpoint
 * @param a Konfigurasjon for responsiv oppførsel
 * @param breakpoint Nåværende breakpoint
 * @returns true dersom config skal brukes for breakpointet
 */
const isValidForCurrentBreakpoint = (config: BreakpointConfig, breakpoint: Breakpoint): boolean =>
  Breakpoint[config.breakpoint] >= breakpoint;

/**
 * Finn konfigurasjon for nåværende breakpoint
 * @param config Konfigurasjon for responsiv oppførsel
 * @param breakpoint Nåværende breakpoint
 * @returns Konfigurasjon for responsiv oppførsel
 */
const getConfigForBreakpoint = (config: BreakpointConfig | BreakpointConfig[], breakpoint: Breakpoint): BreakpointConfig | undefined => {
  if (Array.isArray(config)) {
    config.sort(sortByBreakpointsDescending);

    return config.find(x => isValidForCurrentBreakpoint(x, breakpoint));
  } else if (config && isValidForCurrentBreakpoint(config, breakpoint)) {
    return config;
  }
};

/**
 * Finn konfigurasjon for nåværende breakpoint og tabellbredde
 * @param config Konfigurasjon for responsiv oppførsel
 * @param breakpoint Nåværende breakpoint
 * @param tableWidth Bredde på tabell i px
 * @returns Konfigurasjon for responsiv oppførsel
 */
export const getCurrentConfig = (
  config: BreakpointConfig | BreakpointConfig[],
  breakpoint: Breakpoint,
  tableWidth: number,
  windowWidth: number
): BreakpointConfig | undefined => {
  const breakpointConfig = getConfigForBreakpoint(config, breakpoint);
  const canUseHorizontalScroll = isTouchDevice();
  const canUseCenteredOverflow = tableWidth <= windowWidth;

  if (!breakpointConfig) {
    return;
  }

  if (
    breakpointConfig.variant === SmallViewportVariant.centeredoverflow &&
    !canUseCenteredOverflow &&
    breakpointConfig.fallbackVariant === SmallViewportVariant.horizontalscroll
  ) {
    return {
      variant: canUseHorizontalScroll ? SmallViewportVariant.horizontalscroll : SmallViewportVariant.none,
      breakpoint: breakpointConfig.breakpoint,
    };
  }

  if (
    breakpointConfig.variant === SmallViewportVariant.centeredoverflow &&
    !canUseCenteredOverflow &&
    breakpointConfig.fallbackVariant !== SmallViewportVariant.centeredoverflow
  ) {
    return {
      variant: breakpointConfig.fallbackVariant ?? SmallViewportVariant.none,
      breakpoint: breakpointConfig.breakpoint,
    };
  }

  if (
    breakpointConfig.variant === SmallViewportVariant.horizontalscroll &&
    !canUseHorizontalScroll &&
    breakpointConfig.fallbackVariant === SmallViewportVariant.centeredoverflow
  ) {
    return {
      variant: canUseCenteredOverflow ? SmallViewportVariant.centeredoverflow : SmallViewportVariant.none,
      breakpoint: breakpointConfig.breakpoint,
    };
  }

  if (
    breakpointConfig.variant === SmallViewportVariant.horizontalscroll &&
    !canUseHorizontalScroll &&
    breakpointConfig.fallbackVariant !== SmallViewportVariant.horizontalscroll
  ) {
    return {
      variant: breakpointConfig.fallbackVariant ?? SmallViewportVariant.none,
      breakpoint: breakpointConfig.breakpoint,
    };
  }

  return breakpointConfig;
};

/**
 * Finn klassenavn for responsiv oppførsel
 * @param config Konfigurasjon for responsiv oppførsel
 * @returns Klassenavn
 */
export const getBreakpointClass = (config?: BreakpointConfig): string | undefined =>
  config && configUsesCss(config) ? mapConfigToClass(config) : undefined;

/** @deprecated Midlertidig, fjernes i v3.0.0 */
export const isOldFormat = (x: unknown): x is SmallViewportVariant =>
  (x as SmallViewportVariant) === SmallViewportVariant.block || (x as SmallViewportVariant) === SmallViewportVariant.horizontalscroll;

/** @deprecated Midlertidig, fjernes i v3.0.0 */
export const getBackwardsCompatibleConfig = (variant: SmallViewportVariant): BreakpointConfig => {
  return {
    breakpoint: 'sm',
    variant,
  };
};
