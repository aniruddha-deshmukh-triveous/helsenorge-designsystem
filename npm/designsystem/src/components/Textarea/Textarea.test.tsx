import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Textarea from './Textarea';
import { FormMode } from '../../constants';

import * as uuidUtils from '../../utils/uuid';
jest.spyOn(uuidUtils, 'uuid').mockReturnValue(`-unik-id`);

describe('Gitt at Textarea skal vises', (): void => {
  describe('Når Textarea rendres', (): void => {
    test('Så vises Textarea', (): void => {
      const { container } = render(<Textarea label={'Skriv din historie'} />);

      expect(container).toMatchSnapshot();

      const label = screen.getByText('Skriv din historie');
      expect(label).toBeVisible();

      const input = screen.getByRole('textbox');
      expect(input).toBeVisible();
      expect(input.className).toBe('content-wrapper__input');
    });
  });

  describe('Når disabled er true', (): void => {
    test('Så vises Checkbox som disabled', (): void => {
      render(<Textarea label={'Check me out!'} disabled />);

      const input = screen.getByRole('textbox');
      expect(input).toBeDisabled();
    });
  });

  describe('Når FormMode er onblueberry', (): void => {
    test('Så vises Textarea med onblueberry styling', (): void => {
      render(<Textarea label={'Check me out!'} mode={FormMode.onblueberry} />);

      const contentWrapper = screen.getByRole('textbox').parentElement;
      expect(contentWrapper.className).toBe('content-wrapper content-wrapper--on-blueberry');
    });
  });
  describe('Når mode er ondark', (): void => {
    test('Så vises Textarea med ondark styling', (): void => {
      render(<Textarea label={'Check me out!'} mode={FormMode.ondark} />);

      const contentWrapper = screen.getByRole('textbox').parentElement;
      expect(contentWrapper.className).toBe('content-wrapper content-wrapper--on-dark');
    });
  });

  describe('Når textarea endres med ny tekst og antall tegn skal vises', (): void => {
    test('Så endres teksten og antall tegn', (): void => {
      render(<Textarea label={'Skriv din historie her'} maxCharacters={50} />);

      const textarea1 = screen.getByLabelText('Skriv din historie her');

      expect(screen.getByText('0/50 tegn')).toBeVisible();

      fireEvent.change(textarea1, { target: { value: 'Jeg tester teksten her.' } });

      const input = screen.getByRole('textbox');

      expect(input).toHaveValue('Jeg tester teksten her.');
      expect(screen.getByText('23/50 tegn')).toBeVisible();
    });
  });

  describe('Når textarea har en default value', (): void => {
    test('Så skal default verdien vises', (): void => {
      render(<Textarea label={'Skriv din historie her'} defaultValue="Min historie" />);
      const input = screen.getByRole('textbox');
      expect(input).toHaveValue('Min historie');
    });
  });

  describe('Når textarea har en placeholder value', (): void => {
    test('Så skal placeholder verdien vises', (): void => {
      render(<Textarea label={'Skriv din historie her'} placeholder="Fin placeholder" />);
      const input = screen.getByPlaceholderText('Fin placeholder');
      expect(input).toBeVisible();
    });
  });

  describe('Når textarea har en gått over maksgrensen', (): void => {
    test('Så skal textarea vise at input er ugyldig', (): void => {
      render(<Textarea label={'Skriv din historie her'} grow maxCharacters={10} />);

      const textarea1 = screen.getByLabelText('Skriv din historie her');

      expect(screen.getByText('0/10 tegn')).toBeVisible();

      fireEvent.change(textarea1, { target: { value: 'Jeg tester teksten her.' } });

      expect(screen.getByText('23/10 tegn')).toBeVisible();

      const contentWrapper = screen.getByRole('textbox').parentElement;

      expect(contentWrapper.className).toEqual('content-wrapper content-wrapper--invalid');
    });
  });
});
