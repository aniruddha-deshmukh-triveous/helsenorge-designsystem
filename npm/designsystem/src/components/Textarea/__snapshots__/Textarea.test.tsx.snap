// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at Textarea skal vises Når Textarea rendres Så vises Textarea 1`] = `
<div>
  <div
    class="textarea"
    data-analyticsid="textarea"
  >
    <div
      class="textarea__label-wrapper"
    >
      <label
        for="-unik-id"
      >
        Skriv din historie
      </label>
    </div>
    <div
      class="content-wrapper"
    >
      <textarea
        aria-invalid="false"
        class="content-wrapper__input"
        id="-unik-id"
        rows="3"
      />
    </div>
  </div>
</div>
`;
